public/app/core/components/Upgrade/UpgradeBox.tsx

```diff
@@ -88,11 +79,150 @@ const getUpgradeBoxStyles = (theme: GrafanaTheme2, size: ComponentSize) => {
         outline: 2px solid ${theme.colors.primary.main};
       }
     `,
-    buttonSecondary: css`
-      color: ${theme.colors.text.secondary};
-    `,
     icon: css`
       margin: ${theme.spacing(0.5, 1, 0.5, 0.5)};
     `,
   };
 };
+
+export interface UpgradeContentProps {
+  image: string;
+  featureUrl?: string;
+  featureName: string;
+  description?: string;
+  listItems: string[];
+  caption?: string;
+}
+
+export const UpgradeContent = ({
+  listItems,
+  image,
+  featureUrl,
+  featureName,
+  description,
+  caption,
+}: UpgradeContentProps) => {
+  const styles = useStyles2(getUpgradeContentStyles);
+  return (
+    <div className={styles.container}>
+      <div className={styles.content}>
+        <h3 className={styles.title}>Get started with {featureName}</h3>
+        {description && <h6 className={styles.description}>{description}</h6>}
+        <ul className={styles.list}>
+          {listItems.map((item, index) => (
+            <li key={index}>
+              <Icon name={'check'} size={'xl'} className={styles.icon} /> {item}
+            </li>
+          ))}
+        </ul>
+        {featureUrl && (
+          <LinkButton fill={'text'} href={featureUrl} className={styles.link} target="_blank" rel="noreferrer noopener">
+            Learn more
+          </LinkButton>
+        )}
+      </div>
+      <div className={styles.media}>
+        <img src={getImgUrl(image)} alt={'Feature screenshot'} />
+        {caption && <p className={styles.caption}>{caption}</p>}
+      </div>
+    </div>
+  );
+};
+
+const getUpgradeContentStyles = (theme: GrafanaTheme2) => {
+  return {
+    container: css`
+      display: flex;
+      justify-content: space-between;
+    `,
+    content: css`
+      width: 45%;
+      margin-right: ${theme.spacing(4)};
+    `,
+    media: css`
+      width: 55%;
+
+      img {
+        width: 100%;
+      }
+    `,
+    description: css`
+      color: ${theme.colors.text.primary};
+      font-weight: ${theme.typography.fontWeightLight};
+    `,
+    list: css`
+      list-style: none;
+      margin: ${theme.spacing(4, 0, 2, 0)};
+
+      li {
+        display: flex;
+        align-items: flex-start;
+        color: ${theme.colors.text.primary};
+        padding: ${theme.spacing(1, 0)};
+      }
+    `,
+    icon: css`
+      color: ${theme.colors.success.main};
+      margin-right: ${theme.spacing(1)};
+    `,
+    link: css`
+      margin-left: ${theme.spacing(2)};
```

Comment:
Maybe we should not add this margin when there is no list. It seems a bit off for the usage insights feature and the reporting one (in the Share modal).

Reply:
Both of those components use vertical box component, which doesn't have these styles. Also the link button has a background that appears on hover so if we adjust the text with the description text, it'll be off when hovered.